hugegraph-api/src/main/java/com/baidu/hugegraph/api/auth/ProjectAPI.java

```diff
@@ -0,0 +1,235 @@
+            throw new IllegalArgumentException("Invalid project id: " + id);
+        }
+        project = jsonProject.build(action, project);
+        if (this.isAddGraph(action)) {
+            authManager.updateProjectAddGraph(projectId, jsonProject.graph);
+        } else if (this.isRemoveGraph(action)) {
+            authManager.updateProjectRemoveGraph(projectId, jsonProject.graph);
+        } else {
+            authManager.updateProject(project);
+        }
+        return manager.serializer(g).writeAuthElement(project);
+    }
+
+    @GET
+    @Timed
+    @Produces(APPLICATION_JSON_WITH_CHARSET)
+    public String list(@Context GraphManager manager,
+                       @PathParam("graph") String graph,
+                       @QueryParam("limit") @DefaultValue("100") long limit) {
+        LOG.debug("Graph [{}] list project", graph);
+
+        HugeGraph g = graph(manager, graph);
+        List<HugeProject> projects = manager.authManager()
+                                            .listAllProject(limit);
+        return manager.serializer(g).writeAuthElements("projects", projects);
+    }
+
+    @GET
+    @Timed
+    @Path("{id}")
+    @Produces(APPLICATION_JSON_WITH_CHARSET)
+    public String get(@Context GraphManager manager,
+                      @PathParam("graph") String graph,
+                      @PathParam("id") String id) {
+        LOG.debug("Graph [{}] get project: {}", graph, id);
+
+        HugeGraph g = graph(manager, graph);
+        HugeProject project;
+        try {
+            project = manager.authManager().getProject(UserAPI.parseId(id));
+        } catch (NotFoundException e) {
+            throw new IllegalArgumentException("Invalid project id: " + id);
+        }
+        return manager.serializer(g).writeAuthElement(project);
+    }
+
+    @DELETE
+    @Timed
+    @Path("{id}")
+    @Consumes(APPLICATION_JSON)
+    public void delete(@Context GraphManager manager,
+                       @PathParam("graph") String graph,
+                       @PathParam("id") String id) {
+        LOG.debug("Graph [{}] delete project: {}", graph, id);
+
+        @SuppressWarnings("unused") // just check if the graph exists
+        HugeGraph g = graph(manager, graph);
+        try {
+            manager.authManager().deleteProject(UserAPI.parseId(id));
+        } catch (NotFoundException e) {
+            throw new IllegalArgumentException("Invalid project id: " + id);
+        }
+    }
+
+    public static boolean isAddGraph(String action) {
+        return ACTION_ADD_GRAPH.equals(action);
+    }
+
+    public static boolean isRemoveGraph(String action) {
+        return ACTION_REMOVE_GRAPH.equals(action);
+    }
+
+    private static class JsonProject implements Checkable {
+
+        @JsonProperty("project_description")
+        private String description;
+        @JsonProperty("project_name")
+        private String name;
+        @JsonProperty("project_graph")
+        private String graph;
```

Comment:
graphs

Reply:
Currently only one graph needs to be updated or deleted, so here is singular

Reply:
keep consistency between server response and client request